scriptlets/clean_machine

```diff
@@ -0,0 +1,73 @@
+#!/bin/bash
+# this is a generic clean script that deletes any thing that may break the
+# current test run due to a previous run. This is especially useful for
+# noprovision machines.
+
+# We can do all these things without checking the result as we may be running
+# in any situation, so the previous run could have installed a snap, deb, source
+# etc.
+
+# First lets check that launching this was intentional
+
+if [ "$1" = "--im-sure" ]; then
+  echo "Cleaning the machine"
+else
+  echo "This is dangerous as it tries to purge an environment before running"
+  echo "the tests. You should not launch this on your own machine."
+  echo "if you are sure launch: `basename "$0"` --im-sure"
+  exit 1
+fi
+
+set +e
+
+# delete all checkbox snaps, this includes any frontend or custom frontend
+# (they all contain checkbox in their name). This should also purge any
+# running checkbox*agent service
+CHECKBOX_SNAPS=$(snap list | \grep -E 'checkbox' | \grep -P -o "^[\w-]+")
```

Comment:
Enlighten me: why `\grep` and not plain `grep`?

Reply:
I have a lot of aliases on my machine, I will fix this, I was copy pasting from my console

Reply:
Fixed